src/modules/jj.rs

```diff
@@ -0,0 +1,288 @@
+    )?;
+    let parsed = JJStatus::new(&output.stdout)?;
+    let formatter = match StringFormatter::new(config.format) {
+        Ok(formatter) => formatter,
+        Err(e) => {
+            log::error!("Error in parsing format string for jj module {}", e);
+            return None;
+        }
+    };
+    let formatted = formatter
+        .map_meta(|s, _| match s {
+            "symbol" => Some(config.symbol),
+            "divergent_symbol" => {
+                if parsed.divergent == "true" {
+                    Some(config.divergent_symbol)
+                } else {
+                    None
+                }
+            }
+            "no_description_symbol" => {
+                if parsed.description.is_empty() {
+                    Some(config.no_description_symbol)
+                } else {
+                    None
+                }
+            }
+            _ => None,
+        })
+        .map_style(|s| {
+            Some(Ok(match s {
+                "change_id_prefix_style" => config.change_id_prefix_style,
+                "change_id_suffix_style" => config.change_id_suffix_style,
+                "commit_id_prefix_style" => config.commit_id_prefix_style,
+                "commit_id_suffix_style" => config.commit_id_suffix_style,
+                "branch_style" => config.branch_style,
+                _ => return None,
+            }))
+        })
+        .map(|v| parsed.format(v))
+        .parse(None, Some(context));
+    let formatted = match formatted {
+        Ok(formatted) => formatted,
+        Err(e) => {
+            log::error!("Error in parsing format string for jj module {}", e);
+            return None;
+        }
+    };
+    module.set_segments(formatted);
+
+    Some(module)
+}
+
+fn jj_template(len: u8) -> String {
+    let template = format!(
+        r##"change_id.shortest({len}).prefix() ++ "#," ++ change_id.shortest({len}).rest() ++ "#," ++ commit_id.shortest({len}).prefix() ++ "#," ++ commit_id.shortest({len}).rest() ++ "#," ++ divergent ++ "#," ++ description ++ "#," ++ branches.join(",")"##,
+    );
+    template
+}
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+struct JJStatus<'s> {
+    change_id_prefix: &'s str,
+    change_id_suffix: &'s str,
+    commit_id_prefix: &'s str,
+    commit_id_suffix: &'s str,
+    divergent: &'s str,
+    description: &'s str,
+    branches: &'s str,
+}
+
+impl<'s> JJStatus<'s> {
+    fn new(input: &'s str) -> Option<JJStatus<'s>> {
+        let parsed: Vec<&str> = input.split("#,").collect();
+        if parsed.len() != 7 {
+            log::error!("expected at least 5 parts, got {}", parsed.len());
+            return None;
+        }
+        let (
+            change_id_prefix,
+            change_id_suffix,
+            commit_id_prefix,
+            commit_id_suffix,
+            divergent,
+            description,
+            branches,
+        ) = (
+            parsed[0], parsed[1], parsed[2], parsed[3], parsed[4], parsed[5], parsed[6],
+        );
```

Comment:
the whole check here could be done using `let else`:

```rust
let [var1, var2] = parsed else {
    // log error
    return None;
};
```

or even `match`:

```rust
match parsed {
    [var1, var2] => Some(S { var1, var2 }),
    _ => {
        // log error
        None
    }
}
```

Reply:
done